exercises/concept/the-farm/.docs/instructions.md

```diff
@@ -69,3 +52,24 @@ fodder, err := DivideFood(twentyFodderNoError, -5)
 // fodder == 0.0
 // err.Error() == "silly nephew, there cannot be -5 cows"
 ```
+
+## 5. Handle ErrScaleMalfunction and other errors
```

Comment:
Doing this part last might reflect your implementation but it is not good practice. To decide whether to return the negative fodder error in task 2, you need to look at `fodder`. That is only "allowed" when you first ensured that `err` is `nil`. If the error is not nil, you should never blindly use the result of a function call, it can contain whatever random value.

`ErrScaleMalfunction` is an exception as the instructions that that is ok to work with the result but we should probably change that at some point so it does not set a bad example anymore.

Reply:
> `ErrScaleMalfunction` is an exception as the instructions that that is ok to work with the result but we should probably change that at some point so it does not set a bad example anymore.

Absolutely. In fact, the case of `ErrScaleMalfunction` is the one that led me to the incorrectly conclude on the prior handling of values before error. As per instructions, `ErrScaleMalfunction` error requires the implementation to double the fodder amount and return a nil error. However, if the fodder amount is negative then this rule is bypassed by the `Negative fodder` case (with or without an error) implying that inspection of the value should be given precendence.